aurorawatch-pipeline/aurora_transform.py

```diff
@@ -0,0 +1,49 @@
+'''Script to transform the extracted data from the AuroraWatch UK API into a format
+ready to be loaded.'''
+
+import logging
+from datetime import datetime as dt
+
+from aurora_extract import extract
+
+
+ALERT_COLOURS = {'Green': 1, 'Yellow': 2, 'Amber': 3, 'Red': 4}
+
+
+def get_colour_id(colour: str) -> int:
+    '''Returns the colour id for a given colour.'''
+    if not isinstance(colour, str):
+        raise TypeError('The colour must be given in string format.')
+
+    colour_id = ALERT_COLOURS.get(colour.strip().title())
+
+    if colour_id:
+        logging.info('Colour ID found.')
+        return colour_id
+
+    err_msg = f'No colour ID found for colour: {colour}'
+    logging.error(err_msg)
```

Comment:
Good use of logging here, so of our other tests were a little light on logging, so it's a important inclusion here. Especially as the primary thing we are fetching from this API is the colour indicator, it's a big deal if we can't fetch it for whatever reason.